templates/components/multiagent/typescript/workflow/type.ts

```diff
@@ -1,11 +1,36 @@
-import { WorkflowEvent } from "@llamaindex/core/workflow";
+import { WorkflowEvent } from "@llamaindex/workflow";
+import { MessageContent } from "llamaindex";
 
 export type AgentInput = {
-  message: string;
+  message: MessageContent;
   streaming?: boolean;
 };
 
+export type AgentRunEventType = "text" | "progress";
+
+export type ProgressEventData = {
+  id: string;
+  total: number;
+  current: number;
+};
+
+export type AgentRunEventData = ProgressEventData;
+
 export class AgentRunEvent extends WorkflowEvent<{
-  name: string;
-  msg: string;
-}> {}
+  agent: string;
+  text: string;
+  type: AgentRunEventType;
+  data?: AgentRunEventData;
+}> {
+  constructor(options: {
+    agent: string;
+    text: string;
+    type?: AgentRunEventType;
+    data?: AgentRunEventData;
+  }) {
+    super({
+      ...options,
+      type: options.type || "text",
+    });
+  }
+}
```

Comment:
how about removing the constructor and setting type `text` in the code?
```suggestion
}> {}
```